### PR TITLE
[amp-addthis]Fix floating tool css issue and add unit-test

### PR DESCRIPTION
**Pull request by taojing10**
1. add a eventlistener for addthis message and check if it is a floating tool, if yes give a position css style
2. add unit test for floating tool

### CODE REVIEW

File: extensions/amp-addthis/0.1/amp-addthis.js [hunk 1]
```diff
@@ -174,6 +175,30 @@ class AmpAddThis extends AMP.BaseElement {
       // events.
       this.setupListeners_({ampDoc, loc, pubId: this.pubId_});
     }
+
+    //add addition eventListener for iframe postMessage to handle floating tool
+    this.win.addEventListener('message', function(event) {
+      if (event.origin !== ORIGIN || !getData(event)) {
+        return;
+      }
+      const addThisConfig = tryParseJson(getData(event));
```
Comment (aghassemi) [inline extensions/amp-addthis/0.1/amp-addthis.js]: `handleAddThisMessage` also already parses the data.

File: extensions/amp-addthis/0.1/amp-addthis.js [hunk 2]
```diff
@@ -174,6 +175,30 @@ class AmpAddThis extends AMP.BaseElement {
...
+      if (addThisConfig
+        && hasOwn(addThisConfig,'event')
+        && addThisConfig['event'] === CONFIGURATION_EVENT) {
+        if (hasOwn(addThisConfig, 'config')
+          && hasOwn(addThisConfig['config'], 'widgets')) {
+          for (const key in addThisConfig['config']['widgets']) {
+            if (hasOwn(addThisConfig['config']['widgets'][key],'id')
+              && addThisConfig['config']['widgets'][key]['id'] === 'shfs'
+              && document.getElementById(key)) {
+              const style = 'width:100%;height:100%;position:fixed;bottom:0px';
```
Comment (aghassemi) [inline extensions/amp-addthis/0.1/amp-addthis.js]: I think I can see what this is doing, but there is a UX problem here. By the time the component loads and makes it here, it is too late to make such style changes without causing relayout which is not allowed in AMP without user-interaction.

The floating element should already be given this style change by the page author ahead of time.

Comment (aghassemi) [reply extensions/amp-addthis/0.1/amp-addthis.js]: (aside from the UX problem, this also would behave badly in iOS in the AMP Viewer. We specially case position:fixed elements to solve iOS janky scrolling issue, doing it this way will not allow us to discover this).

Comment (taojing10) [reply extensions/amp-addthis/0.1/amp-addthis.js]: Hi @aghassemi  The `position:fixed` won't fix this problem, what if we add these css to iframe before our tool rendered?

Comment (aghassemi) [reply extensions/amp-addthis/0.1/amp-addthis.js]: If you add it to `iframe`, the `amp-addthis` would still be taking space wherever it was inserted as its layout is fixed.

Ideal solution is either for page author to place it in a position fixed element if they wish for `amp-addthis` to be floating or for `amp-addthis` to support a `floating` attribute which authors add like `<amp-sticky-ad layout="nodisplay" floating>`

**Pull request closed without merging**